### ISSUE TITLE
Red question mark for images taken by camera only on iPhones

### ISSUE THREAD

**Issue by riocallos** [yasirkula/UnityNativeGallery#217]
I'm experiencing a similar red question mark issue like this one [https://github.com/yasirkula/UnityNativeGallery/issues/160](https://github.com/yasirkula/UnityNativeGallery/issues/160) but under different conditions.

It only happens for images taken by the camera on iPhones with iOS 14. It works fine when importing downloaded pictures on your phone.

I don't have an iPhone with me so I cannot directly debug. I only have an iPad with latest iPadOS version and I'm not experiencing this issue.

This is the code snippet I where I get the image path.
`#elif UNITY_IOS || UNITY_ANDROID
                if( NativeGallery.IsMediaPickerBusy() )
                    return;
                NativeGallery.Permission permission = NativeGallery.GetImageFromGallery( ( path ) =>
                {
                    Debug.Log( "Image path: " + path );
                    if(path != null && path != "" )
                    {
                        StartCoroutine(UploadPhoto("file://" + path, null));
                    }

                }, "Upload Photo", "image/*" );
#endif`

Note that this works well for Android and the cases mentioned above. I only have a problem for pictures taken by the iPhone's camera. I already have the latest library version.

**Comment by yasirkula** [comment]
If you aren't using NativeGallery.LoadImageAtPath, I'd highly recommend you to do so. It uses native functions and can load all OS-supported image formats (in theory, at least). Unity's built-in LoadImage function can handle only PNG and JPEG files. It is possible that iOS 14 camera saves images in another format like HEIC.

**Comment by riocallos** [comment]
> If you aren't using NativeGallery.LoadImageAtPath, I'd highly recommend you to do so. It uses native functions and can load all OS-supported image formats (in theory, at least). Unity's built-in LoadImage function can handle only PNG and JPEG files. It is possible that iOS 14 camera saves images in another format like HEIC.

I am not directly loading the image. I am only getting the file contents then uploading it to my server then I'm displaying from there. Thank you I think I need to convert that HEIC format to png or jpg.

**Comment by yasirkula** [comment]
Yeah converting the image would be the safest solution (you need to set _markTextureNonReadable_ parameter to _false_). As a plus, you can shrink the image to a desired size if, for example, 512px is sufficient for your use-cases and you don't want to upload a 4k image at full size. You can use _maxSize_ parameter for that.

**Comment by UsmanKhanLodhi** [comment]
Hey @riocallos ,

Did you find a solution for converting HEIC files to JPG? If so, I'd really appreciate it if you could share it with me. Thanks!

**Comment by yasirkula** [comment]
@UsmanKhanLodhi `NativeGallery.LoadImageAtPath("insert HEIC image path", markTextureNonReadable: false).EncodeToJPG()`

**Comment by UsmanKhanLodhi** [comment]
@yasirkula  We're encountering an issue where the texture returned by NativeGallery.LoadImageAtPath() is always null for HEIC images.

`public void PickImage()
    {
        NativeGallery.Permission permission = NativeGallery.GetImageFromGallery((path) =>
        {
          
            ImageName = Path.GetFileName(path);
            ImageExtension = Path.GetExtension(path);
           
            
            if (path != null)
            {
                Texture2D texture = null;

                if (ImageExtension == ".heic")
                {
                    print("Handle HEIC file format here");
                    texture = NativeGallery.LoadImageAtPath(path, 2556, false, true);
                    

                }
                else
                {
                    print("Handle Regular file format here");
                    texture = NativeGallery.LoadImageAtPath(path, 2556, false, true);
                }
                print("Textue= " + texture);

                if (texture == null)
                {
                    
                    return;
                }
            
            
        }
        });
    }`
    
 It always gives us a null texture. Additionally, I have attached the HEIC image file link: https://github.com/tigranbs/test-heic-images/blob/master/image1.heic.

Please review it and solve this issue because most phones generate HEIC files. Therefore, the ability of the app to handle the HEIC format is a mandatory condition.

Thank you.

**Comment by yasirkula** [comment]
For the future readers, related Issue: #316
